data/tutorial/backend.md

```diff
@@ -254,3 +267,30 @@ module KV: Irmin.KV_MAKER = functor (C: Irmin.Contents.S) ->
     (Irmin.Branch.String)
     (Irmin.Hash.SHA1)
 ```
+We also have to provide a configuration for our backend specifying the parameters needed when initialising a store. In our example, we start with an empty configuration, which comes with  `root` as a parameter. We can then instantiate the store and create a repo:
+```ocaml
+let config ?(config = Irmin.Private.Conf.empty) ?root () =
+  let module C = Irmin.Private.Conf in
+  C.add config C.root root
+
+module Store = KV (Irmin.Contents.String)
+let repo = Store.Repo.v (config ())
```

Comment:
Is it now possible to set/get a key in this repo without getting an assertion error? Perhaps it's worth adding a line that does that, just so we don't get a regression in future.

Reply:
The issue was not in the configuration, but in the `add` function: the hash needs to applied on `Irmin.Type.to_bin_string V.t value` and not on `Irmin.Type.to_string V.t value`. 

The empty configuration works here, but I though it is cleaner to show to how to create a configuration anyway. I'm not sure if we should write this in the tutorial, I think it might be confusing, no?

Reply:
I agree that creating a configuration is good, and that should be here. I was thinking about adding an _extra_ few lines to show the backend actually being used, if only because that check would have could the bug with the backend (assertion failure in `tree.ml` when running the mdx tests). However, it's perhaps not worth having in the tutorial.

More generally, it would be nice to have a solution for testing the objects in these tutorials without those tests needing to exist in the tutorial output; perhaps @samoht's suggestion of a diff/promote workflow would fit here.